/**
 * Copyright 2022 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package agent;

message AgentRegReq {
  uint32 node_rank = 1;
  uint32 local_worker_num = 2;
}

message AgentRegRsp {
  bool success = 1;
}

message CheckClusterStatusReq {
  uint32 node_rank = 1;
  uint32 local_worker_num = 2;
}

message CheckClusterStatusRsp {
  bool ready = 1;
  repeated uint32 assigned_rank_list = 2;
}

service AgentRegService {
  rpc Register (AgentRegReq) returns (AgentRegRsp);
  rpc CheckClusterStatus (CheckClusterStatusReq) returns (CheckClusterStatusRsp);
}


